File: 2022/day6.go
package main

import (
	"io/ioutil"
	"log"
)

func unique(arr string) bool {
    m := make(map[rune]bool)
    for _, i := range arr {
        _, ok := m[i]
        if ok {
            return false
        }

        m[i] = true
    }

    return true
}

func main() {
	content, err := ioutil.ReadFile("./2022/day6input.txt")
	MARKEK_LEN := 12
	input := string(content)

	if err != nil {
		log.Fatal(err)
	}

	for i := 0; i < len(input) - MARKEK_LEN; i++ {
		if (unique(string(input[i:i+MARKEK_LEN]))) {
			print(i+MARKEK_LEN)
			break
		}
	}
}

File: 2022/day6.py
MARKER_LENGTH = 14

def main():
    f = open('./2022/day6input.txt', 'r')
    lines = f.readlines()

    for line in lines:
        for i in range(len(line) - MARKER_LENGTH):
            substring = line[i:i+MARKER_LENGTH]

            if len(set(substring)) == MARKER_LENGTH:
                print(substring)
                print(i+MARKER_LENGTH)
                break

    return

if __name__ == "__main__":
    main()
